core/src/main/java/bisq/core/payment/payload/PaymentMethod.java

```diff
@@ -258,30 +243,39 @@ public static PaymentMethod fromProto(PB.PaymentMethod proto) {
     // API
     ///////////////////////////////////////////////////////////////////////////////////////////
 
-    // Used for dummy entries in payment methods list (SHOW_ALL)
-    public PaymentMethod(String id) {
-        this(id, 0, Coin.ZERO);
-    }
-
     public static PaymentMethod getPaymentMethodById(String id) {
-        return getAllValues().stream()
+        return paymentMethods.stream()
                 .filter(e -> e.getId().equals(id))
                 .findFirst()
                 .orElseGet(() -> new PaymentMethod(Res.get("shared.na")));
     }
 
-    // Hack for SF as the smallest unit is 1 SF ;-( and price is about 3 BTC!
     public Coin getMaxTradeLimitAsCoin(String currencyCode) {
-        if (currencyCode.equals("SF") || currencyCode.equals("BSQ"))
+        // Hack for SF as the smallest unit is 1 SF ;-( and price is about 3 BTC!
+        if (currencyCode.equals("SF"))
             return Coin.parseCoin("4");
+
+        // We use the class field maxTradeLimit only for mapping the risk factor.
+        long riskFactor;
+        if (maxTradeLimit == DEFAULT_TRADE_LIMIT_VERY_LOW_RISK.value)
```

Comment:
There is a problem that if a low risk payment method (e.g. altcoin) has maxTradeLimit already set to two BTC it will be set as `riskFactor` of 8 as the `DEFAULT_TRADE_LIMIT_VERY_LOW_RISK.value` is 1 BTC.

Reply:
Ups, the PaymentMethod constructor was called from the Offer, that caused the issue. I will remove that. the fields are all final now so it has to be guaranteed anyway that the data from the offer payload matches the default values in the code. The difference is that we supported with that that an offer could have different limits and durations as the actual code base (in case a new version has changed those). I think that is not much an issue as we must not change those default values anyway and if so they only may increase otherwise we could render an offer invalid (if amount becomes lower as limit). The maker gets a "promis" when he creates the offer to those data (most relevant is trade duration here) so we have to consider that if we would decide to change default values. So far there have been no need to change those but for altcoins it might become interesting to support very short durations (live trades) in some future releases. But lets discuss once we get there, the duration should be probably refactored then so that there is only the default value in and the real value will be delivered by some service, or we add a new payment methods reflecting that new properties (easier probably).